src/test/scala/beam/analysis/physsim/PickUpDropOffCollectorTest.scala

```diff
@@ -0,0 +1,219 @@
+package beam.analysis.physsim
+
+import beam.physsim.{LinkPickUpsDropOffs, PickUpDropOffCollector, TimeToValueCollection}
+import beam.utils.{BeamVehicleUtils, EventReader}
+import org.scalatest.{FunSuite, Matchers}
+
+import scala.collection.mutable
+
+class PickUpDropOffCollectorTest extends FunSuite with Matchers {
+
+  test("TimeToLinkCollection should work as expected") {
+    val ttcc = new TimeToValueCollection()
+    ttcc.getClosestTimeIndex(42) shouldBe None
+    ttcc.getClosestValue(42, 100000) shouldBe None
+
+    ttcc.addTimeToValue(100, 3)
+    ttcc.getClosestTimeIndex(42) shouldBe Some(0)
+    ttcc.getClosestValue(42, 100 - 42) shouldBe Some(3)
+    ttcc.getClosestValue(42, 100 - 43) shouldBe None
+
+    ttcc.addTimeToValue(200, 2)
+    ttcc.getClosestValue(149, 50) shouldBe Some(3)
+    ttcc.getClosestValue(152, 50) shouldBe Some(2)
+
+    ttcc.addTimeToValue(300, 10)
+    ttcc.getClosestValue(149, 50) shouldBe Some(3)
+    ttcc.getClosestValue(151, 50) shouldBe Some(2)
+    ttcc.getClosestValue(151, 48) shouldBe None
+    ttcc.getClosestValue(249, 50) shouldBe Some(2)
+    ttcc.getClosestValue(251, 50) shouldBe Some(10)
+    ttcc.getClosestValue(251, 48) shouldBe None
+  }
+
+  test("BEAMVILLE. Should read expected link pick-ups and drop-offs") {
+    val eventsFilePath = "test/test-resources/PickUpsDropOffsCollectorTestData/beamville.events.xml"
+    val vehicleTypeFilePath = "test/test-resources/PickUpsDropOffsCollectorTestData/beamville.vehicleTypes.csv"
+
+    val events = EventReader.fromXmlFile(eventsFilePath)
+
+    val vehicleTypes = BeamVehicleUtils.readBeamVehicleTypeFile(vehicleTypeFilePath)
+    val collector = new PickUpDropOffCollector(vehicleTypes)
+
+    var linkToPickUpDropOff: mutable.HashMap[Int, LinkPickUpsDropOffs] = null
+
+    collector.notifyIterationStarts(null)
+    events.foreach(event => collector.handleEvent(event))
+    linkToPickUpDropOff = collector.getLinkToPickUpsDropOffs
+
+    linkToPickUpDropOff shouldNot be(null)
+
+    val (pickUps, dropOffs) = linkToPickUpDropOff.foldLeft((0, 0)) {
+      case ((pickUps, dropOffs), (_, pickUpDropOff)) =>
+        val linkPickUps: Int = pickUpDropOff.timeToPickUps.values.sum
+        val linkDropOffs: Int = pickUpDropOff.timeToDropOffs.values.sum
+
+        (pickUps + linkPickUps, dropOffs + linkDropOffs)
+    }
+
+    pickUps shouldBe (19 + 20)
+    dropOffs shouldBe (19 + 20)
+
+    /*
```

Comment:
Can this be dropped - is there a reason to keep commented out code?

Reply:
this is a python code I used to calculate values from the events file, I am not sure if it will be better to put this piece of code elsewhere because it is only needed to obtain values to generate test code.

Reply:
I am fine with the comment given what you said